### ISSUE TITLE
Vue.js setup problems

### ISSUE THREAD

**Issue by Azahliil** [Ergosign/storybook-addon-pseudo-states#34]
Hey there!

We're currently checking out Storybook as an alternative for our previous solution, and my dear friend our designer asked me to check out if it was easily possible to display the different states of our components:
So thats how we got here :D

I now added the configs according to the docs, but postCSS seems to have some trouble processing the styles.

As I'm not familiar with postCSS or building with webpack that much I thought I might ask if you could help me :)

### Using 
Storybook 6.1.2
Vue 2.6.11
storybook-addon-pseudo-states 0.0.1-alpha.40
Contents of the ./storybook/main.js and postcss.config.js can be found here: https://pastebin.com/wTt40fVC

I apologize in advance if I just made a noob mistake, but any help on this would be highly appreciated! 

### Error Message
```
ERROR in ./src/stories/button.css (./node_modules/@storybook/core/node_modules/css-loader/dist/cjs.js??ref--10-1!./node_modules/@storybook/core/node_modules/postcss-loader/src??postcss!./node_modules/style-loader/dist/cjs.js!./node_modules/css-loader/dist/cjs.js??ref--14-1!./node_modules/postcss-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js??ref--14-3!./src/stories/button.css)
Module build failed (from ./node_modules/@storybook/core/node_modules/postcss-loader/src/index.js):
SyntaxError

(1:1) Unknown word

> 1 | import api from "!../../node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js";
    | ^
  2 |             import content from "!!../../node_modules/css-loader/dist/cjs.js??ref--14-1!../../node_modules/postcss-loader/dist/cjs.js!../../node_modules/sass-loader/dist/cjs.js??ref--14-3!./button.css";
  3 | 
```

Also I noticed that the documentation for Vue tells you to add the angular addon, which is probably not correct :D
(https://github.com/Ergosign/storybook-addon-pseudo-states/blob/master/src/vue/README.md)

**Comment by andreasphil** [comment]
Hi @Azahliil,

thanks for your feedback. We're aware that the Vue integration is still pretty rough, so your problems are likely not due to any mistakes on your end. But we've got some updates and improved docs coming up soon (probably later this week). I'll keep you updated!

**Comment by Azahliil** [comment]
Oh that sounds awesome!

Thank you for the quick response - Excited for what you have been cooking up :D

**Comment by andreasphil** [comment]
Hi @Azahliil

we just released a new version `0.0.1-alpha.41` and updated the [readme](https://github.com/Ergosign/storybook-addon-pseudo-states/tree/master/src/vue). As the `alpha` indicates it's all still very much experimental at this point but feel free to give it a go. There are some limitations but the basic use case should work.

And let us know how it went 😊

**Comment by Azahliil** [comment]
Awesome!

I jumped at it right away :D 

The docs were at least a lot less confusing this way, and postcss seems to be doing it's thing, but the sass-loader now acts up :D

```
ERROR in ./src/components/animations/hp-loader/hp-loader.vue?vue&type=style&index=0&lang=scss& (./node_modules/css-loader/dist/cjs.js!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/sass-loader/dist/cjs.js!./node_modules/css-loader/dist/cjs.js??ref--14-1!./node_modules/sass-loader/dist/cjs.js??ref--14-2!./node_modules/postcss-loader/dist/cjs.js??ref--14-3!./node_modules/vue-loader/lib??vue-loader-options!./src/components/animations/hp-loader/hp-loader.vue?vue&type=style&index=0&lang=scss&)
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: expected "{".
  ╷
2 │ import ___CSS_LOADER_API_SOURCEMAP_IMPORT___ from "../../../../node_modules/css-loader/dist/runtime/cssWithMappingToString.js";
  │  
```

If you have any idea how to fix that I would be forever grateful!
My Webpack customization in `main.js` currently looks like this, I assume something with the loaders or their order is off (But thats just a wild guess)

```js
webpackFinal: async (config, {configType}) => {
        config.resolve.alias = {
            'vue$': 'vue/dist/vue.esm.js',
            '@': path.resolve(__dirname, "../src")
        };
        config.module.rules.push({
            test: /\.(scss|css)$/,
            use: [
                {
                    loader: 'style-loader',
                },
                {
                    loader: 'css-loader',
                    options: {
                        // ATTENTION when using css modules
                        modules: {
                            // !!! must not use [hash]'
                            localIdentName: '[path][name]__[local]',
                        },
                    },
                },
                {
                    loader: "sass-loader",
                    options: {
                        additionalData: "@import '@/styles/main-lib'; @import '@/styles/main-styleguide';",
                    },
                },
                {
                    loader: 'postcss-loader',
                    options: {
                        postcssOptions: {
                            config: path.resolve(__dirname, "postcss.config.js"),
                        },
                    },
                },
            ]
        });
```

And I hope I don't bother you with my problems, I did notice that little `alpha` in the version number 😅  - But I hope I can make the experience for the ones that follow after me much smoother and maybe help a little with this amazing project this way :3

**Comment by philippone** [comment]
Hi @Azahliil ,

could you try to swap `postcss-loader` and `sass-loader` .

Not sure if this helps, but it is worth a try.

**Comment by Azahliil** [comment]
I just realized I didn't even need the `style-loader` and `css-loader` anymore, removing those fixed the `sass-loader` errors!
And putting `postcss-loader` before `sass-loader` worked! The styles now get applied just how they should!

Someone will be very happy when I tell the good news on monday in our weekly :D

Just one little thing, currently, as I just saw, the classes only get applied in the canvas view.
The docs view doesn't get the pseudo classes. But I don't know if thats just something weird going on with the mdx format. 

So far a big **THANK YOU** To the both of you!
And if it wont work in docs view, thats something I could very well live with.

**Comment by andreasphil** [comment]
Glad to hear it's working for you!

We noticed the behavior in the docs section too. We're not sure about the cause but we've got a few ideas to work around it. However we can't give you an ETA yet as to when these fixes are going to be released. For now we've added it as a known limitation to the readme.

To be honest I haven't worked with the docs tab a lot, so if you notice anything that might cause or even solve the problem we'd appreciate your feedback!
